src/lang/lang_types.ml

```diff
@@ -666,12 +667,20 @@ let rec bind a0 b =
             (function
               | Ord ->
                   let rec check b =
-                    let b = demeth b in
+                    let m, b = split_meths b in
                     match b.descr with
                       | Ground _ -> ()
                       | EVar (j, c) ->
                           if List.mem Ord c then ()
                           else b.descr <- EVar (j, Ord :: c)
+                      | Tuple [] ->
+                          (* For records, we want to ensure that all fields are ordered. *)
```

Comment:
What does that entail?

Reply:
For a record to be orderable, all its fields have to be orderable. For other values, as explained above, we don't compare the fields, so we don't check for orderability (and again, they are functions in general, so they would not be orderable).